cpp/src/sort/segmented_sort.cu

```diff
@@ -89,8 +218,42 @@ std::unique_ptr<column> segmented_sorted_order_common(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr)
 {
+  if (keys.num_rows() == 0 || keys.num_columns() == 0) {
+    return cudf::make_empty_column(type_to_id<size_type>());
+  }
+
   CUDF_EXPECTS(segment_offsets.type() == data_type(type_to_id<size_type>()),
                "segment offsets should be size_type");
+
+  if (not column_order.empty()) {
+    CUDF_EXPECTS(static_cast<std::size_t>(keys.num_columns()) == column_order.size(),
+                 "Mismatch between number of columns and column order.");
+  }
+
+  if (not null_precedence.empty()) {
+    CUDF_EXPECTS(static_cast<std::size_t>(keys.num_columns()) == null_precedence.size(),
+                 "Mismatch between number of columns and null_precedence size.");
+  }
+
+  // the average row size for which to prefer fast sort
+  constexpr cudf::size_type MAX_AVG_LIST_SIZE_FOR_FAST_SORT{100};
+  // the maximum row count for which to prefer fast sort
+  constexpr cudf::size_type MAX_LIST_SIZE_FOR_FAST_SORT{1 << 18};
+
+  // fast-path for single column sort:
+  // - single-column table
+  // - not stable-sort
+  // - no nulls and allowable fixed-width type
+  // - size and width are limited -- based on benchmark results
+  if (keys.num_columns() == 0 and sorting == sort_method::UNSTABLE and
```

Comment:
`keys.num_columns() == 0`
equals zero?
How are unit tests and benchmarks not catching this?

Reply:
Good catch. This is a mistake and was meant only to capture the baseline benchmarks..